src/components/tasks/card/ProgressText.tsx

```diff
@@ -0,0 +1,30 @@
+import { FC } from 'react';
+import classNames from '@/components/tasks/card/card.module.scss';
+import { handleProgressTextProps } from '@/interfaces/task.type';
+
+const HandleProgressText: FC<handleProgressTextProps> = ({
+    progress,
+    handleSaveProgressUpdate,
+    handleProgressUpdate,
+}) => {
+    if (progress) {
+        return (
+            <div
+                className={classNames.changeProgressText}
+                onClick={() => handleSaveProgressUpdate()}
+            >
+                save Progress
+            </div>
+        );
+    }
+    return (
+        <div
+            className={classNames.changeProgressText}
+            onClick={() => handleProgressUpdate()}
+        >
+            Progress update
+        </div>
+    );
```

Comment:
- the `className` and `onClick` have the same functionality, can we not just add a ternary for the text?

```js
{progress ? "save progress" : "progress update"} 
```

Reply:
but save Progress and progress update are calling different functions whose functionality is different, I had thought about this but then how will the logic work as suggested by you